### PR TITLE
Add keep alive support

### PR DESCRIPTION
**Pull request by vkuptcov**

### CODE REVIEW

**Comment by kulikov** [comment]
+1

File: src/main/java/net/spy/memcached/MemcachedConnection.java [hunk 1]
```diff
@@ -335,7 +336,9 @@ protected void registerMetrics() {
       MemcachedNode qa = connectionFactory.createMemcachedNode(sa, ch, bufSize);
       qa.setConnection(this);
       int ops = 0;
-      ch.socket().setTcpNoDelay(!connectionFactory.useNagleAlgorithm());
+      Socket socket = ch.socket();
+      socket.setTcpNoDelay(!connectionFactory.useNagleAlgorithm());
+      socket.setKeepAlive(!connectionFactory.getKeepAlive());
```
Comment (drcrallen) [inline src/main/java/net/spy/memcached/MemcachedConnection.java]: Why is this negated?

**Comment by youngm** [comment]
I'd like to take over this PR and get it integrated.  Is anyone from couchbase still watching this repo and will support me in getting this PR completed?

**Comment by youngm** [comment]
Hello @drcrallen or anyone from Couchbase is this project alive at all?  Is anyone around to accept this PR if I complete it?

**Comment by ingenthr** [comment]
Yes, I can help with that @youngm

**Comment by youngm** [comment]
@ingenthr Great!  I'll review the code and hopefully have and updated PR next week.  Thanks!

**Comment by youngm** [comment]
This PR can be closed.

**Pull request closed without merging**